src/ariel-os-embassy/Cargo.toml

```
@@ -130,20 +130,23 @@ eth-stm32 = ["ariel-os-hal/eth-stm32", "net", "eth"]
 ble = [
   "dep:trouble-host",
   "ariel-os-embassy-common/ble",
-  "ariel-os-hal/ble",
-  # Random is needed for BLE initialization and Address generation
-  "ariel-os-random/csprng",
+  # Some HALs need hwrng to initialize the BLE stack.
   "hwrng",
-  "random",
+  "ariel-os-hal/ble",
 ]
 ble-cyw43 = ["ble", "ariel-os-hal/ble-cyw43"]
 ble-peripheral = ["ble", "ariel-os-hal/ble-peripheral"]
 ble-central = ["ble", "ariel-os-hal/ble-central"]
 
-threading = ["dep:ariel-os-threads", "ariel-os-hal/threading"]
+threading = [
+  "dep:ariel-os-threads",
+  "ariel-os-hal/threading",
+  "embassy-time/generic-queue-128",
```

Going with the maximum here (128) for lack of precise information.

---

Ah, you need a question mark for that, otherwise this also enables `embassy-time`:

```suggestion
  "embassy-time?/generic-queue-128",
```